src/main/java/com/example/cosmocats/common/jpa/NaturalIdType.java

```diff
@@ -0,0 +1,34 @@
+package com.example.cosmocats.common.jpa;
+
+import java.security.SecureRandom;
+import java.time.Year;
+import java.util.function.Supplier;
+
+public enum NaturalIdType {
```

Comment:
Ok, I will give you etxra point for this because I love the effort, and I see you vision but it's just used for a bit different purpose in the real life :) +1